qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,335 @@
+        self._penalty = penalty  # type: float
+
+        return self._dst
+
+    def _conversion_matrix(self, constraint) -> np.ndarray:
+        """Construct conversion matrix for special constraint.
+
+        Returns:
+            Return conversion matrix which is used to construct
+            penalty term in main function.
+
+        """
+
+        vars_dict = constraint.linear.to_dict(use_name=True)
+        vrs = list(vars_dict.items())
+        rhs = constraint.rhs
+        sense = constraint.sense
+
+        num_vars = len(vrs)
+        combinations = list(itertools.combinations(np.arange(num_vars), 2))
+
+        # conversion matrix
+        conv_matrix = np.zeros((num_vars + 1, num_vars + 1), dtype=int)
+
+        for combination in combinations:
+            index1 = combination[0] + 1
+            index2 = combination[1] + 1
+
+            if rhs in (1, num_vars - 1):
+                if sense == ConstraintSense.GE:
+                    conv_matrix[0][0] = conv_matrix[0][0] + 1
+                    conv_matrix[0][index1] = conv_matrix[0][index1] - 1
+                    conv_matrix[0][index2] = conv_matrix[0][index2] - 1
+                conv_matrix[index1][index2] = 1
+            elif rhs == 0:
+                if sense == ConstraintSense.GE:
+                    if vrs[index1 - 1][1] < 0.0:
+                        conv_matrix[0][index1] = 1
+                    elif vrs[index2 - 1][1] < 0.0:
+                        conv_matrix[0][index2] = 1
+                elif sense == ConstraintSense.LE:
+                    if vrs[index1 - 1][1] > 0.0:
+                        conv_matrix[0][index1] = 1
+                    elif vrs[index2 - 1][1] > 0.0:
+                        conv_matrix[0][index2] = 1
+
+                conv_matrix[index1][index2] = -1
+
+        return conv_matrix
+
+    def _is_special_constraint(self, problem, constraint) -> bool:
+        """Determine if constraint is special or not.
+
+        Returns:
+            True: when constraint is special
+            False: when constraint is not special
+        """
+
+        params = constraint.linear.to_dict()
+        num_vars = len(params)
+        rhs = constraint.rhs
+        sense = constraint.sense
+        coeff_array = np.array(list(params.values()))
+
+        # Binary parameter?
+        if not all(problem.variables[i].vartype == Variable.Type.BINARY for i in params.keys()):
+            return False
+
+        if num_vars == 2:
+            if rhs == 1:
+                if all(i == 1 for i in params.values()):
+                    if sense in (Constraint.Sense.LE, Constraint.Sense.GE):
+                        # x+y<=1
+                        # x+y>=1
+                        return True
+            if rhs == 0:
+                if sense in (Constraint.Sense.LE, Constraint.Sense.GE):
+                    # x-y<=0
+                    # x-y>=0
+                    return coeff_array.min() == -1.0 and coeff_array.max() == 1.0
+        elif num_vars > 2:
```

Comment:
If you change this condition to `num_vars >= 2`, you can eliminate `num_vars == 2 and rhs == 1` because they are special cases of `num_vars >= 2`.

Reply:
When changing to`num_vars >= 2`, we need to check `num_vars == 2` in the code block again to identify `x+y>=1`. Also to keep code readability, we'd like to leave as it is.

Reply:
Why do you need to separately check `num_vars == 2 and x + y >= 1`? I think line 248 matches for such case.

Reply:
Thank you for your comment. I've simplified the conditions as you suggested.